src/lib.rs

```diff
@@ -503,6 +505,26 @@ impl GLRenderer
             .get_webgl2_context(viewport_size_pixels)
     }
 
+    /// Creates a `GLRenderer` from callback's result.
+    ///
+    /// The parameter `viewport_size_pixels` should be set to
+    /// the initial canvas size, however this can be changed later using
+    /// [GLRenderer:: set_viewport_size_pixels()].
+    #[cfg(any(doc, doctest, target_arch = "wasm32"))]
+    pub fn new_for_web_canvas_from_callback<V, S>(
+        viewport_size_pixels: V,
+        webgl2_context_cb: S
+    ) -> Result<Self, BacktraceError<GLRendererCreationError>>
+    where
+        V: Into<UVec2>,
+        S: FnOnce() -> WebGl2RenderingContext
```

Comment:
This creates a public dependency on the `web_sys` crate. Would it be possible to create a new feature for this in `Cargo.toml` please?

Reply:
Why public? i see that  on wasm32 target arch this dep is included with WebGl2RenderingContext feature . So it is working only on wasm32.

Reply:
Sorry, I was unclear. I meant that this exposes `WebGl2RenderingContext` in the public Speedy2D API, which forces us to continue depending on `web-sys` in the future (to avoid breaking the code of existing Speedy2D users). If we ever want to replace `web-sys` with another library, we'd have to release a breaking v3.0.0 update.

I'm okay with this, as long as the `use_existing_webgl2_context` API is gated behind a feature. For example:

```toml
[features]
default = ["windowing", "image-loading"]
windowing = ["glutin"]
image-loading = ["image"]
web-sys-features = []
```

```rust
#[cfg(any(feature = "web-sys-features", doc, doctest))]
pub fn use_existing_webgl2_context(
    ...
```

Then the `use_existing_webgl2_context` function will only be enabled if `web-sys-features` is set.